vertx-pojo-mapper-common/src/main/java/de/braintags/vertx/jomnigate/dataaccess/query/impl/AbstractQueryExpression.java

```diff
@@ -93,7 +92,7 @@ private Object handleSingleValue(Object value) {
     } else if (value instanceof Enum) {
       transformedValue = ((Enum<?>) value).name();
     } else if (value instanceof GeoSearchArgument) {
-      transformedValue = new JsonObject(Json.encode(value));
+      transformedValue = JsonObject.mapFrom(value);
     } else {
       try {
         // can not use datastore object mapper here because only the JSON datastore has the object mapper
```

Comment:
the mysql search expression for Geo differs a lot from the one of mongo, but ok, we will change that later together with the refactoring of mysql